File: csharp/pangram/Pangram.cs
using System.Collections.Generic;

public static class Pangram
{
    public static bool IsPangram(string input)
    {
        HashSet<char> lettersFound = new HashSet<char>();

        foreach (char c in input)
            if (char.IsLetter(c))
                lettersFound.Add(char.ToLower(c));

        return lettersFound.Count == 26;
    }
}

File: haskell/pangram/src/Pangram.hs
module Pangram (isPangram) where

import           Data.Char (toLower)
import qualified Data.Set as S

isPangram :: String -> Bool
isPangram text = S.null $
    S.fromAscList ['a'..'z']
        `S.difference`
    S.fromList (map toLower text)
